src/datatypes/dtype.rs

```diff
@@ -129,19 +129,27 @@ impl DataType {
     #[inline]
     pub fn is_numeric(&self) -> bool {
         match self {
-            DataType::Int8
-            | DataType::Int16
-            | DataType::Int32
-            | DataType::Int64
-            | DataType::UInt8
-            | DataType::UInt16
-            | DataType::UInt32
-            | DataType::UInt64
-            // DataType::Float16
-            | DataType::Float32
-            | DataType::Float64 => true,
-            _ => false
-        }
+             DataType::Int8
+             | DataType::Int16
+             | DataType::Int32
+             | DataType::Int64
+             | DataType::UInt8
+             | DataType::UInt16
+             | DataType::UInt32
+             | DataType::UInt64
+             // DataType::Float16
+             | DataType::Float32
+             | DataType::Float64 => true,
+             _ => false
+         }
+    }
+
+    #[inline]
+    pub fn is_castable(&self, cast_to: &DataType) -> DaftResult<bool> {
```

Comment:
This should probably just be a bool rather than a DaftResult. In any case we throw an error, it wouldn't be castable.

Reply:
Ah nice, that makes sense yes